docs/user_guide/hdl_coding_guideline.rst

```diff
@@ -0,0 +1,960 @@
+   // Copyright (C) year-year Analog Devices, Inc. All rights reserved.
+   //
+   // In this HDL repository, there are many different and unique modules, consisting
+   // of various HDL (Verilog or VHDL) components. The individual modules are
+   // developed independently, and may be accompanied by separate and unique license
+   // terms.
+   //
+   // The user should read each of these license terms, and understand the
+   // freedoms and responsabilities that he or she has by using this source/core.
+   //
+   // This core is distributed in the hope that it will be useful, but WITHOUT ANY
+   // WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+   // A PARTICULAR PURPOSE.
+   //
+   // Redistribution and use of source or resulting binaries, with or without modification
+   // of this file, are permitted under one of the following two license terms:
+   //
+   //   1. The GNU General Public License version 2 as published by the
+   //      Free Software Foundation, which can be found in the top level directory
+   //      of this repository (LICENSE_GPL2), and also online at:
+   //      <https://www.gnu.org/licenses/old-licenses/gpl-2.0.html>
+   //
+   // OR
+   //
+   //   2. An ADI specific BSD license, which can be found in the top level directory
+   //      of this repository (LICENSE_ADIBSD), and also on-line at:
+   //      https://github.com/analogdevicesinc/hdl/blob/master/LICENSE_ADIBSD
+   //      This will allow to generate bit files and not release the source code,
+   //      as long as it attaches to an ADI device.
+   //
+   // ***************************************************************************
+   // ***************************************************************************
+
+   'timescale 1ns/100ps
+
+   module prescaler #(
+     // range = 1-16
+     parameter FIRST_PARAMETER = 8,
+     // range = N/A
+     parameter SECOND_PARAMETER = 12
+   ) (
+     input           core_32m_clk,         // 32 MHz clock
+     input           system_clk,           // system clock
+     input           scan_mode_test,       // scan mode clock
+     input           reset_n,              // active low hard reset, synch w/
+                                           // system_clk
+     output  reg     div16_clk,            // input clock divided by 16
+     output  reg     div16_clk_n           // input clock divided by 16 and inverted
+   );
+     // local parameters
+
+     // registers declarations
+
+     reg     [3:0]   count;          // 4-bit counter to make clock divider
+     reg     [3:0]   count1;         // 4-bit counter to make clock divider
+
+     // wires declarations
+
+     wire    [3:0]   count1_ns;      // clock divider next state input
+
+     // functions definitions
+
+     // this block updates the internal counter
+     always @(posedge core_32m_clk or negedge reset_n) begin
+       if (!reset_n) begin
+         count <= 4�b0000;
```

Comment:
Replace � with '
No idea why this happened.

Reply:
Done